contracts/SpotV1.sol

```diff
@@ -148,39 +140,42 @@ contract SpotV1 is ERC20Spot {
         uint beforeWithdrawBalance = userInfo.amount;
 
         // sell liquidity
-        // refactor to strategy tokens base calculation
         uint amountToWithdraw = ((amountToBurn * beforeWithdrawBalance)) / beforeWithdrawSupply;
         _unStake(amountToWithdraw);
 
         _sellLiquidity(amountToWithdraw, swap0, swap1, deadline);
 
-        // withdraw tokens swap[0].path[swap.path.length - 1]
-        if (swap0.path[swap0.path.length - 1] == wrapper) {
+        
+
+        // withdraw tokens tokenToWithdraw
+        if (tokenToWithdraw == wrapper) {
             IWrap(wrapper).withdraw(IWrap(wrapper).balanceOf(address(this)));
+            _log(1, tokenToWithdraw, address(this).balance, amountToWithdraw);
             (bool sentRecipient, ) = payable(msg.sender).call{value: address(this).balance}("");
-            require(sentRecipient, "transfer ETH to recipeint failed");
+            require(sentRecipient, "transfer BNB to recipeint failed");
         } else {
-            IERC20(tokenToWithdraw).safeTransfer(msg.sender, IERC20(tokenToWithdraw).balanceOf(address(this)));
+            uint withdrawTokenAmount = IERC20(tokenToWithdraw).balanceOf(address(this));
+            _log(1, tokenToWithdraw, withdrawTokenAmount, amountToWithdraw);
+            IERC20(tokenToWithdraw).safeTransfer(msg.sender, withdrawTokenAmount);
         }
 
         //burn
         _burn(ownerId, amountToBurn);
 
-        _returnRemainder(swap0.path[swap0.path.length - 1]);
+        // _returnRemainder(swap0.path[swap0.path.length - 1]);
     }
 
     function callAny(address payable _addr, bytes memory _data) public payable onlyRDNOwner(msg.sender) returns(bool success, bytes memory data){
         (success, data) = _addr.call{value: msg.value}(_data);
     }
-
+    
     function restake(
         Swap memory swapReward0,
         Swap memory swapReward1,
         uint deadline
-    ) public {
+    ) public onlyActiveRDNOwner(msg.sender) {
         _restake(deadline, swapReward0, swapReward1);
-
-        _returnRemainder(swapReward0.path[0]);
+        // _returnRemainder(swapReward0.path[0]);
```

Comment:
WARNING: tokens may be forever locked. Consider activate _returnRemainder, or at least add "recoverERC20" function. In the current implementation you may call "callAny" with calldata to IERC20.transfer but it's not intuitive.